src/Chainweb/Pact/Backend/RelationalCheckpointer.hs

```diff
@@ -256,3 +264,45 @@ doLookupSuccessful dbenv (TypedHash hash) = runBlockEnv dbenv $ do
         !hsh <- either fail return $ Data.Serialize.decode blob
         return $! T2 (fromIntegral h) hsh
     go _ = fail "impossible"
+
+doGetBlockHistory :: FromJSON v => Db -> BlockHeader -> Domain k v -> IO BlockTxHistory
+doGetBlockHistory dbenv blockHeader d = runBlockEnv dbenv $ do
+  callDb "doGetBlockHistory" $ \db -> do
+    endTxId <- getEndTxId db bHeight (_blockHash blockHeader)
+    startTxId <- getEndTxId db (pred bHeight) (_blockParent blockHeader)
+    history <- queryHistory db (domainTableName d) startTxId endTxId
+    return $! BlockTxHistory $ V.fromList $ M.toList $ foldl groupByTxid mempty history
```

Comment:
> why call insertWith (++) when we can :?
We're using a map to group by so not clear how `:` is relevant

Reply:
Used `foldl'`

Reply:
If we don't care about ordering, then instead of 

```haskell
M.insertWith (++) t [l] r
```

You could 

```haskell
M.insertWith (flip (:)) t l r
```

Reply:
Order isn't necessarily unimportant, and per-txid lists are small ... I'm good to leave it